crates/aranya-client/src/aqc/crypto.rs

```diff
@@ -28,75 +30,106 @@ pub(super) static CTRL_PSK: LazyLock<Arc<PresharedKey>> = LazyLock::new(|| {
 
 #[derive(Debug)]
 pub(crate) struct ServerPresharedKeys {
-    keys: Mutex<HashMap<Vec<u8>, Arc<PresharedKey>>>,
+    keys: Mutex<HashSet<PskIdAsKey>>,
+    identities: Mutex<HashMap<AqcChannelId, Vec<PskIdentity>>>,
+    channels: Mutex<HashMap<PskIdentity, AqcChannelId>>,
```

Comment:
Could just store the related psks:

```suggestion
    keys: Mutex<HashMap<PskIdAsKey, Arc<[Arc<PresharedKey>]>>>,
```

then:
```rust
if let Some(psks) = keys.remove(chosen) {
    for psk in psks.iter() {
        keys.remove(psk.identity());
    }
}
```